openedx/core/djangoapps/user_authn/utils.py

```diff
@@ -125,18 +126,54 @@ def is_safe_login_or_logout_redirect(redirect_to, request_host, dot_client_id, r
     return is_safe_url
 
 
-def generate_password(length=12, chars=string.ascii_letters + string.digits):
-    """Generate a valid random password"""
+def generate_password(length=12, min_digits=1, min_lowercase=1, min_uppercase=1, min_symbols=1, min_special=1):
```

Comment:
@Waleed-Mujahid can we reduce the amount of changes in these core files and only add the minimum code to achieve the fix?

Reply:
If you think it can break in future, then add a method in edly app and simply call that over here without changing too many code lines